lib/network.go

```diff
@@ -4774,6 +4775,161 @@ type TransactionSpendingLimit struct {
 	DAOCoinLimitOrderLimitMap map[DAOCoinLimitOrderLimitKey]uint64
 }
 
+func (tsl *TransactionSpendingLimit) ToMetamaskString(params *DeSoParams) string {
+	var str string
+	var indentationCounter int
+	_indt := _computeIndentation
+
+	str += "Spending limits on the derived key:\n"
+	indentationCounter++
+
+	// GlobalDESOLimit
+	if tsl.GlobalDESOLimit > 0 {
+		str += _indt(indentationCounter) + "Total $DESO Limit: " +
+			strconv.FormatUint(tsl.GlobalDESOLimit, 10) + " $DESO\n"
+	}
+
+	// TransactionCountLimitMap
+	if len(tsl.TransactionCountLimitMap) > 0 {
+		var txnCountStr []string
+		str += _indt(indentationCounter) + "Transaction Count Limit: \n"
+		indentationCounter++
+		for txnType, limit := range tsl.TransactionCountLimitMap {
+			txnCountStr = append(txnCountStr, _indt(indentationCounter)+txnType.String()+": "+
+				strconv.FormatUint(limit, 10)+"\n")
+		}
+		// Ensure deterministic ordering of the transaction count limit strings by doing a lexicographical sort.
+		sort.Strings(txnCountStr)
+		for _, limitStr := range txnCountStr {
+			str += limitStr
+		}
+		indentationCounter--
+	}
+
+	// CreatorCoinOperationLimitMap
+	if len(tsl.CreatorCoinOperationLimitMap) > 0 {
+		var creatorCoinLimitStr []string
+		str += _indt(indentationCounter) + "Creator Coin Operation Limits:\n"
+		indentationCounter++
+		for limitKey, limit := range tsl.CreatorCoinOperationLimitMap {
+			opString := _indt(indentationCounter) + "[\n"
+
+			indentationCounter++
+			opString += _indt(indentationCounter) + "Creator Public Key: " +
```

Comment:
technically this isn't the creator pub key, but rather the PKID. MOST of the time it will be the same, but should the message reflect that it's a PKID?

Reply:
applies to all uses of PKID below as well.